spec/lib/workos/webhooks_spec.rb

```diff
@@ -5,98 +5,70 @@
-        id: 'directory_user_01FAEAJCR3ZBZ30D8BD1924TVG',
-        state: 'active',
+  before(:each) do
+    @payload = File.read("#{SPEC_ROOT}/support/webhook_payload.txt")
+    @secret = 'secret'
+    @timestamp = Time.at(Time.now.to_i * 1000)
+    unhashed_string = "#{@timestamp.to_i}.#{@payload}"
+    digest = OpenSSL::Digest.new('sha256')
+    @signature_hash = OpenSSL::HMAC.hexdigest(digest, @secret, unhashed_string)
+    @expectation = {
+      id: 'directory_user_01FAEAJCR3ZBZ30D8BD1924TVG',
+      state: 'active',
+      emails: [{
+        type: 'work',
+        value: 'blair@foo-corp.com',
+        primary: true,
+      }],
+      idp_id: '00u1e8mutl6wlH3lL4x7',
+      object: 'directory_user',
+      username: 'blair@foo-corp.com',
+      last_name: 'Lunceford',
+      first_name: 'Blair',
+      directory_id: 'directory_01F9M7F68PZP8QXP8G7X5QRHS7',
+      raw_attributes: {
+        name: {
+          givenName: 'Blair',
+          familyName: 'Lunceford',
+          middleName: 'Elizabeth',
+          honorificPrefix: 'Ms.',
+        },
+        title: 'Developer Success Engineer',
+        active: true,
         emails: [{
           type: 'work',
           value: 'blair@foo-corp.com',
           primary: true,
         }],
-        idp_id: '00u1e8mutl6wlH3lL4x7',
-        object: 'directory_user',
-        username: 'blair@foo-corp.com',
-        last_name: 'Lunceford',
-        first_name: 'Blair',
-        directory_id: 'directory_01F9M7F68PZP8QXP8G7X5QRHS7',
-        raw_attributes: {
-          name: {
-            givenName: 'Blair',
-            familyName: 'Lunceford',
-            middleName: 'Elizabeth',
-            honorificPrefix: 'Ms.',
-          },
-          title: 'Developer Success Engineer',
-          active: true,
-          emails: [{
-            type: 'work',
-            value: 'blair@foo-corp.com',
-            primary: true,
-          }],
-          groups: [],
-          locale: 'en-US',
-          schemas: [
-            'urn:ietf:params:scim:schemas:core:2.0:User',
-            'urn:ietf:params:scim:schemas:extension:enterprise:2.0:User'
-          ],
-          userName: 'blair@foo-corp.com',
-          addresses: [{
-            region: 'CO',
-            primary: true,
-            locality: 'Steamboat Springs',
-            postalCode: '80487',
-          }],
-          externalId: '00u1e8mutl6wlH3lL4x7',
-          displayName: 'Blair Lunceford',
-          "urn:ietf:params:scim:schemas:extension:enterprise:2.0:User": {
-            manager: {
-              value: '2',
-              displayName: 'Kathleen Chung',
-            },
-            division: 'Engineering',
-            department: 'Customer Success',
+        groups: [],
+        locale: 'en-US',
+        schemas: [
+          'urn:ietf:params:scim:schemas:core:2.0:User',
+          'urn:ietf:params:scim:schemas:extension:enterprise:2.0:User'
+        ],
+        userName: 'blair@foo-corp.com',
+        addresses: [{
```

Comment:
Are you sure you are not looking to attract a stalker 😛? I'd definitely put something random into the address fields...

Reply:
I think I'm too de-sensitized because I live in a small town where everyone knows everyone 😆